components/dialog/dialog-fullscreen.js

```diff
@@ -129,8 +164,8 @@ class DialogFullscreen extends LocalizeCoreElement(AsyncContainerMixin(DialogMix
 			<div class="d2l-dialog-inner">
 				<div class="d2l-dialog-header">
 					<div>
-						<h2 id="${this._titleId}" class="d2l-heading-3">${this.titleText}</h2>
-						<d2l-button-icon icon="tier1:close-small" text="${this.localize('components.dialog.close')}" @click="${this._abort}"></d2l-button-icon>
+						<h3 id="${this._titleId}" class="d2l-heading-3">${this.titleText}</h3>
```

Comment:
I'm not sure if it's possible to have this icon change from the big icon to the small one when the screen is mobile-sized. Any tips for this? I'm thinking we could put both the small close icon and the big one here and toggle visibility based on screen size.

Reply:
I am assuming the goal is to have a larger button?  To achieve a larger button, I think we need to override these CSS variables that are defined in `d2l-button-icon`.  Depending on the size, it may also be appropriate to bump up the radius for the corners. @geurts fyi... might need specifics.

```css
--d2l-button-icon-min-height
--d2l-button-icon-min-width
--d2l-button-icon-border-radius
```

Ref: https://github.com/BrightspaceUI/core/blob/master/components/button/button-icon.js#L49

Note: the `d2l-button-icon` demo page has a demo (with custom styles) that you can try if you need an example.

Reply:
And those overrides can be applied in a media-query as required to have a different size for different size viewports.  If we want to change the actual icon, then I think things might become slightly more involved since there may be some focus code that looks for the abort button.

Reply:
I believe we do want to change the button as the design doc specifies the thicker version of the close button versus the previous thin version. Not sure if @geurts has a preference if we actually need the thicker button or a larger size would suffice? Let me know 😄

Reply:
I think for larger viewports we want the larger button and thicker icon, which we can accomplish using the icon currently used in this PR, plus the CSS variables.  

However the design mentions that for smaller devices it should be indistinguishable from the regular dialog, which uses a smaller button and the smaller close icon.  There are a variety of ways to accomplish that, but I would use `matchMedia` in combination with conditionally rendering the button with the appropriate icon.

Reply:
So I get how we can use the CSS variables to change the size of the button responsively, but how do with change the icon? I don't think we want two buttons that we hide/show for the reasons you mentioned above around dealing with cases where focus is on one of them.

Reply:
Can the icon string itself (`tier1:close-large-thick`) be stored in property that causes a re-render when it changes (and we adjust it on resize)?  Or will that break with the way icons are dynamically imported as needed?

Reply:
Yeah I think we can just use `matchMedia` (wiring up to its event) to change a property that controls which icon to use.